apollo-router/src/services/layers/query_analysis.rs

```diff
@@ -191,7 +195,38 @@ impl QueryAnalysisLayer {
                 request
                     .context
                     .extensions()
-                    .with_lock(|mut lock| lock.insert::<ParsedDocument>(doc));
+                    .with_lock(|mut lock| lock.insert::<ParsedDocument>(doc.clone()));
+
+                let extended_refs_enabled =
+                    match self.configuration.apollo_plugins.plugins.get("telemetry") {
+                        Some(telemetry_config) => {
+                            match serde_json::from_value::<TelemetryConfig>(
+                                telemetry_config.clone(),
+                            ) {
+                                Ok(conf) => {
+                                    matches!(
+                                        conf.apollo.experimental_apollo_metrics_reference_mode,
+                                        ApolloMetricsReferenceMode::Extended
+                                    )
+                                }
+                                _ => false,
+                            }
+                        }
+                        None => false,
+                    };
```

Comment:
This will create a big perf regression I think. Because we will have to deserialize the config for each request and it's something that can be done by using a downcast and I would suggest to not do this in a hot path like this one. It should be done only when the config is updated and at the creation of plugins/layers.

Reply:
I'll figure out a way to pre-calculate this on startup or something like that.

Reply:
I've updated it to load the config on startup and tested that it works with hot-reload, but I'm not sure what you meant by using a downcast.